src/search/pdbs/pattern_collection_generator_hillclimbing.cc

```diff
@@ -342,14 +349,23 @@ void PatternCollectionGeneratorHillclimbing::hill_climbing(
         while (true) {
             ++num_iterations;
             int init_h = current_pdbs->get_value(initial_state);
-            utils::g_log << "current collection size is "
-                         << current_pdbs->get_size() << endl;
-            utils::g_log << "current initial h value: ";
+            if (verbosity >= utils::Verbosity::VERBOSE) {
+                utils::g_log << "current collection size is "
+                             << current_pdbs->get_size() << endl;
+                utils::g_log << "current initial h value: ";
```

Comment:
no need to repeat the second utils::g_log

Reply:
how expensive is the is_dead_end(state) check? 
If it is cheap, the code would be cleaner, if you directly chain:
` << (pdbs->is_dead_end(state)? "inftiy": iinit_h << endl;`
Then, the next if block is simplified and the else block can be removed.

Reply:
It is an iteration over all patterns/PDBs and a lookup for each. I could simply store a boolean to avoid recomputation and simplify the output as suggested.